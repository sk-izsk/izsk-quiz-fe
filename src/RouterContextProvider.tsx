import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';

export interface RouterContextProviderProps {}

const RouterContextProvider: React.FC<RouterContextProviderProps> = ({ children }) => {
  const history: any = createMemoryHistory();
  return <Router history={history}>{children} </Router>;
};

export { RouterContextProvider };
