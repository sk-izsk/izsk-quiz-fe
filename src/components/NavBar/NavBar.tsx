import { AppBar, IconButton, makeStyles, Toolbar, Typography } from '@material-ui/core';
import clsx from 'clsx';
import React, { FC, useState } from 'react';
import { FcAbout, FcHome, FcIdea } from 'react-icons/fc';
import { IoMdMenu } from 'react-icons/io';
import { MdAccountCircle } from 'react-icons/md';
import { RiLoginCircleLine, RiLogoutCircleRLine } from 'react-icons/ri';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { Drawer } from '..';
import { Account } from '../../redux/accountSlice';
import { RootState } from '../../redux/store';
import { theme } from '../../theme/muiTheme';

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  hide: {
    display: 'none',
  },
  link: {
    textDecoration: 'none',
    color: theme.palette.warning.contrastText,
    '&:hover': {
      textDecoration: 'none',
    },
  },
}));

export interface MenuItem {
  name: string;
  to: string;
  icon: JSX.Element;
  display?: boolean;
}

const menuItems: (display: boolean) => MenuItem[] = (display: boolean) => [
  {
    name: 'Login',
    to: '/login',
    icon: <RiLoginCircleLine size={25} color={theme.palette.primary.main} />,
    display: !display,
  },
  {
    name: 'Sign Up',
    to: '/sign-up',
    icon: <MdAccountCircle size={25} color={theme.palette.primary.main} />,
    display: !display,
  },
  {
    name: 'Home',
    to: '/home',
    icon: <FcHome size={25} />,
    display,
  },
  {
    name: 'Start quiz !!!',
    to: '/quiz',
    icon: <FcIdea size={25} />,
    display,
  },
  {
    name: 'Logout',
    to: '/login',
    icon: <RiLogoutCircleRLine size={25} color={theme.palette.secondary.main} />,
    display,
  },
  {
    name: 'About',
    to: '/about',
    icon: <FcAbout size={25} />,
    display: true,
  },
];

const NavBar: FC = () => {
  const classes = useStyles();
  const [open, setOpen] = useState<boolean>(false);
  const userDetails: Account = useSelector<RootState, Account>((state: RootState) => state.account);

  const handleDrawer: () => void = () => {
    setOpen(!open);
  };

  return (
    <div className={classes.root}>
      <AppBar
        position='fixed'
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open,
        })}
        color='primary'
      >
        <Toolbar>
          <IconButton
            color='inherit'
            aria-label='open drawer'
            onClick={handleDrawer}
            edge='start'
            className={clsx(classes.menuButton, open && classes.hide)}
          >
            <IoMdMenu />
          </IconButton>
          <Typography variant='h6'>
            <Link className={classes.link} to='/'>
              iZsk Quiz
            </Link>
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer open={open} handleDrawer={handleDrawer} menuItems={menuItems(userDetails.isLoggedIn)} />
    </div>
  );
};

export { NavBar };
