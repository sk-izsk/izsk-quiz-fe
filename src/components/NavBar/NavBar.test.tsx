import { render } from '@testing-library/react';
import React from 'react';
import { RouterContextProvider } from '../../RouterContextProvider';
import { NavBar } from './NavBar';

test('NavBar renders correctly', () => {
  const { asFragment } = render(
    <RouterContextProvider>
      <NavBar />
    </RouterContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
