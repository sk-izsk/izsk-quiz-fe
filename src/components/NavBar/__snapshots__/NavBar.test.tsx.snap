// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 384h384v-42.666H64V384zm0-106.666h384v-42.667H64v42.667zM64 128v42.665h384V128H64z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
        >
          <a
            class="makeStyles-link-6"
            href="/"
          >
            iZsk Covid Tracker
          </a>
        </h6>
      </div>
    </header>
  </div>
   
</DocumentFragment>
`;
